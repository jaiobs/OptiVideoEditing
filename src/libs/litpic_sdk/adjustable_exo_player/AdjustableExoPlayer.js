import React, {Component} from 'react';
import {
  View,
  requireNativeComponent,
  findNodeHandle,
  NativeEventEmitter,
  NativeModules,
} from 'react-native';

const AdjustableExoVideoPlayer = requireNativeComponent('AdjustableExoVideoPlayer');
const AdjustableExoVideoPlayerManager = NativeModules.AdjustableExoVideoPlayerModule;

export default class AdjustableExoPlayer extends Component {
  constructor(props) {
    super(props);
    const eventEmitter = new NativeEventEmitter(AdjustableExoVideoPlayerManager);
    this.eventListener = eventEmitter.addListener(
      'EventLandscapeVideoCropping',
      event => {
        console.log(event);
        this.props.cropPosition(event.xPosition);
      },
    );
  }

  seekTo(event){
    console.log("adjustable player start position- ",event);
    AdjustableExoVideoPlayerManager.seekToPosition(this._trimmerHandle, event);
  }

  updatePreview(event){
    console.log("adjustable player update preview ",event);
    AdjustableExoVideoPlayerManager.updatePreviewVideo(this._trimmerHandle, event);
  }

  releaseListeners(){
    AdjustableExoVideoPlayerManager.releaseListeners(this._trimmerHandle);
  }

  _setReference = (ref: ?Object) => {
    if (ref) {
      this._trimmerRef = ref;
      this._trimmerHandle = findNodeHandle(ref);
    } else {
      this._trimmerRef = null;
      this._trimmerHandle = null;
    }
  };

  render() {
    return (
      <View style={{flex: 1}}>
        <AdjustableExoVideoPlayer ref={this._setReference} {...this.props} />
      </View>
    );
  }
}
