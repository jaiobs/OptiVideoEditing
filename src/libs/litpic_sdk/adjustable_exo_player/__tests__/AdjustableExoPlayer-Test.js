import React from 'react';
import renderer from 'react-test-renderer';
import AdjustableExoPlayer from '../AdjustableExoPlayer';

describe("AdjustableExoPlayer snapshot test",() => {
    it("It should render AdjustableExoPlayer Correctly", () =>{
        const tree = renderer.create(
            <AdjustableExoPlayer></AdjustableExoPlayer>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
});