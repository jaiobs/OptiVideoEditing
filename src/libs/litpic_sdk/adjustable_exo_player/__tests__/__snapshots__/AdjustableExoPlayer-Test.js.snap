// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdjustableExoPlayer snapshot test It should render AdjustableExoPlayer Correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <AdjustableExoVideoPlayer />
</View>
`;
